### PR TITLE
Add clarification around low watermark documentation.

### LINKED ISSUE #48104: [DOC] Clarification around low disk watermark documentation
**Documentation enhancement request**:

Currently the documentation says:

> `cluster.routing.allocation.disk.watermark.low`
> Controls the low watermark for disk usage. It defaults to 85%, meaning that Elasticsearch will not allocate shards to nodes that have more than 85% disk used. It can also be set to an absolute byte value (like 500mb) to prevent Elasticsearch from allocating shards if less than the specified amount of space is available. This setting has no effect on the primary shards of newly-created indices or, specifically, any shards that have never previously been allocated.

The `This setting has no effect on the primary shards of newly-created indices or, specifically, any shards that have never previously been allocated.` part is quite confusing. In fact, it has effect on replica shards as these can't be allocated when low watermark is tripped (even on newly-created indices).

**To sum up:**
- primary shards of newly created indices are unaffected and will be allocated.
- replica shards of newly created indices are affected and won't be allocated.
- re-allocated / re-balanced shards won't be allocated to the node.

## Enhancement

Change: 
`This setting has no effect on the primary shards of newly-created indices or, specifically, any shards that have never previously been allocated.`

To something like this: 
`This setting has no effect on the primary shards of newly-created indices but replicas won't be allocated. Existing shards can't be re-allocated or re-balanced to a node with low watermark.`

## Test for a newly-created indice (shards allocation)

> [instance-0000000001] low disk watermark [60gb] exceeded on [TLltk6a9SQOnjKgJndoHkQ][instance-0000000001][/app/data/nodes/0] free: 59.8gb[99.6%], replicas will not be assigned to this node

```
PUT replicalloc4
{
  "settings": {
    "number_of_replicas": 1
  }
}

GET _cat/shards?v&index=replicalloc*
replicalloc4 0     p      STARTED       0  230b 10.46.47.236 instance-0000000002
replicalloc4 0     r      UNASSIGNED
```

### Test for existing shard re-allocation

```
POST /_cluster/reroute
{
  "commands" : [
      {
          "move" : {
              "index" : "replicalloc4", "shard" : 0,
              "from_node" : "instance-0000000002", "to_node" : "instance-0000000001"
          }
      }
 ]
}
```

output:
> "[move_allocation] can't move 0, from {instance-0000000002}{ec9jxAyqQpeQi55_LGhBKg}{nrzEfkTnS2muUQ9tslDOxg}{10.46.47.236}{10.46.47.236:19287}{dim}{logical_availability_zone=zone-1, server_name=instance-0000000002.b3ec28ce2c9a4c2a85e517d28973d6da, availability_zone=asia-northeast1-b, xpack.installed=true, region=unknown-region, instance_configuration=gcp.data.highio.1}, to {instance-0000000001}{TLltk6a9SQOnjKgJndoHkQ}{nPxoZShVStm_gNQ9p1WTBg}{10.46.47.234}{10.46.47.234:19763}{dim}{logical_availability_zone=zone-0, server_name=instance-0000000001.b3ec28ce2c9a4c2a85e517d28973d6da, availability_zone=asia-northeast1-a, xpack.installed=true, instance_configuration=gcp.data.highio.1, region=unknown-region}, since its not allowed, reason: [YES(shard has no previous failures)][YES(shard is primary and can be allocated)][YES(explicitly ignoring any disabling of allocation due to manual allocation commands via the reroute API)][YES(can relocate primary shard from a node with version [7.4.0] to a node with equal-or-newer version [7.4.0])][YES(no snapshots are currently running)][YES(ignored as shard is not being recovered from a snapshot)][YES(node passes include/exclude/require filters)][YES(the shard does not exist on the same node)][NO(the node is above the low watermark cluster setting [cluster.routing.allocation.disk.watermark.low=60gb], having less than the minimum required [60gb] free space, actual free: [59.7gb])][YES(below shard recovery limit of outgoing: [0 < 2] incoming: [0 < 2])][YES(total shard limits are disabled: [index: -1, cluster: -1] <= 0)][YES(node meets all awareness attribute requirements)]"

➡️ `[NO(the node is above the low watermark cluster setting [cluster.routing.allocation.disk.watermark.low=60gb], having less than the minimum required [60gb] free space, actual free: [59.7gb])]`

Any thoughts on this? Or suggestion for better wording?
I would be happy to make a PR for that.

### PR DESCRIPTION
**Pull request by renshuki**
Fixes #48104

### CODE REVIEW

File: docs/reference/modules/cluster/disk_allocator.asciidoc [hunk 1]
```diff
@@ -20,8 +20,8 @@ file or updated dynamically on a live cluster with the
     85% disk used. It can also be set to an absolute byte value (like `500mb`)
     to prevent Elasticsearch from allocating shards if less than the specified
     amount of space is available. This setting has no effect on the primary
-    shards of newly-created indices or, specifically, any shards that have
-    never previously been allocated.
+    shards of newly-created indices but replicas won't be allocated. Existing
```
Comment (DaveCTurner) [inline docs/reference/modules/cluster/disk_allocator.asciidoc]:
```suggestion
    shards of newly-created indices but will prevent their replicas from being allocated.
```

File: docs/reference/modules/cluster/disk_allocator.asciidoc [hunk 2]
```diff
@@ -20,8 +20,8 @@ file or updated dynamically on a live cluster with the
...
+    shards can't be re-allocated or re-balanced to a node with low watermark.
```
Comment (DaveCTurner) [inline docs/reference/modules/cluster/disk_allocator.asciidoc]:
```suggestion
```

**Review by DaveCTurner — COMMENTED** [review]
I left a suggestion. I think we should not talk about reallocation or rebalancing here, see https://github.com/elastic/elasticsearch/issues/48104#issuecomment-542618468.

**Review by DaveCTurner — APPROVED** [review]
LGTM

**Comment by renshuki** [comment]
Thanks for the review on this.
LGTM as well, can be merge once checks completed.

**Pull request merged**